WordPress/Classes/Networking/AccountServiceRemoteREST.m

```diff
@@ -151,6 +151,27 @@ - (void)isEmailAvailable:(NSString *)email success:(void (^)(BOOL available))suc
           }];
 }
 
+- (void)checkUsernameAvailability:(NSString *)username
+                            taken:(void (^)())taken
+                        available:(void (^)())available;
+{
+    NSString *path = @"https://public-api.wordpress.com/is-available/username";
+    [self.wordPressComRestApi GET:path
+                       parameters:@{ @"q": username, @"format": @"json"}
+                          success:^(id responseObject, NSHTTPURLResponse *httpResponse) {
+                              if (taken) {
+                                  taken();
```

Comment:
I think we'd still want to check the payload to confirm its contents rather than assume.  For instance, if the endpoint was updated to return a proper JSON response rather than a simple "true" if the name was available.